### ISSUE TITLE
Passenger 5.0.28: Support binary PassengerAgent not found

### ISSUE THREAD

**Issue by john-999** [phusion/passenger#1829]
During my deployment process, Capistrano issues the command...

    $ /usr/local/rvm/bin/rvm ruby-2.3.1 do passenger-config restart-app /home/developer --ignore-app-not-running

...which generates the error message...

```
SSHKit::Runner::ExecuteError: Exception while executing as developer@example.com: passenger-config exit status: 1
passenger-config stdout: *** ERROR: Phusion Passenger doesn't seem to be running. If you are sure that it
is running, then the causes of this problem could be one of:

 1. You customized the instance registry directory using Apache's
    PassengerInstanceRegistryDir option, Nginx's
    passenger_instance_registry_dir option, or Phusion Passenger Standalone's
    --instance-registry-dir command line argument. If so, please set the
    environment variable PASSENGER_INSTANCE_REGISTRY_DIR to that directory
    and run this command again.
 2. The instance directory has been removed by an operating system background
    service. Please set a different instance registry directory using Apache's
    PassengerInstanceRegistryDir option, Nginx's passenger_instance_registry_dir
    option, or Phusion Passenger Standalone's --instance-registry-dir command
    line argument.
passenger-config stderr: Nothing written
```
I know that Passenger is not running, but don't know why. So I attempted to start Nginx manually and it generates the following error message...

```
$ /etc/init.d/nginx start
Starting nginx: nginx: [alert] Unable to start Phusion Passenger: Support binary PassengerAgent not found (tried: /usr/local/rvm/gems/ruby-2.3.1/gems/passenger-5.0.28/buildout/support-binaries/PassengerAgent and /root/.passenger/support-binaries/5.0.28/PassengerAgent). There may be different causes for this:

 - Your 'passenger_root' setting is set to the wrong value. Please see https://www.phusionpassenger.com/library/config/nginx/reference/#passenger_root to learn how to fix the value.
 - The PassengerAgent binary is not compiled. Please run this command to compile it: /usr/local/rvm/gems/ruby-2.3.1/gems/passenger-5.0.28/bin/passenger-config compile-agent
 - Your Passenger installation is broken or incomplete. Please reinstall Passenger. (-1: Unknown error)
nginx.

```
Regarding these 3 points, I can say that my ```passenger_root /usr/local/rvm/gems/ruby-2.3.1/gems/passenger-5.0.28``` definitely seems correct.
Besides that, I have followed the guide at https://www.phusionpassenger.com/library/walkthroughs/deploy/ruby/ownserver/nginx/oss/install_language_runtime.html and everything went smooth (no errors at all).

What is the problem here?

**Comment by FooBarWidget** [comment]
You are using `/etc/init.d/nginx start` to start Nginx. That tells me that there is a high possibility that during the installation you ended up with -two- Nginx installations and that you are actually starting the wrong one.

This page explains it in more detail: https://www.phusionpassenger.com/library/walkthroughs/deploy/ruby/ownserver/nginx/oss/tarball/install_passenger.html#note_about_nginx

Can you check whether this is the case?

**Comment by john-999** [comment]
Thank you for your answer. The issue was actually not a second Nginx installations, I had followed the guide literally and uninstalled the original Nginx installation; and I had added the init.d script manually.

I had to run the ```passenger-config compile-agent``` command, which didn't work right away. It did however after rebooting thereafter. So everything's fine now. (I have no clue why the PassengerAgent binary had apparently not been compiled.)

So I'm closing this issue. Thanks again!
